File: TestProject.NewApplication5.Application/DTOs/ForceUsers_TestService/CreateForceUser.cs

using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Intent.RoslynWeaver.Attributes;
using static TestProject.NewApplication5.Domain.Enums;

[assembly: DefaultIntentManaged(Mode.Fully)]
[assembly: IntentTemplate("Intent.Application.Contracts.DTO", Version = "1.0")]

namespace TestProject.NewApplication5.Application.DTOs.ForceUsers_TestService
{
    [DataContract]
    public class CreateForceUser
    {
        public CreateForceUser()
        {
        }

        public static CreateForceUser Create(
            string name,
            Side side,
            Speciality speciality,
            LightSaberColor lightSaberColor)
        {
            return new CreateForceUser
            {
                Name = name,
                Side = side,
                Speciality = speciality,
                LightSaberColor = lightSaberColor,
            };
        }

        [DataMember]
        public string Name { get; set; }

        [DataMember]
        public Side Side { get; set; }

        [DataMember]
        public Speciality Speciality { get; set; }

        [DataMember]
        public LightSaberColor LightSaberColor { get; set; }
    }
}

File: AngularFrontEnd/ClientApp/src/@generated/dtos/create-force-user.ts
import { LightSaberColor } from './../enums/light-saber-color-enum';
import { Side } from './../enums/side-enum';
import { Speciality } from './../enums/speciality-enum';


//     ____      __             __     ______                           __           __
//    /  _/___  / /____  ____  / /_   / ____/__  ____  ___  _________ _/ /____  ____/ /
//    / // __ \/ __/ _ \/ __ \/ __/  / / __/ _ \/ __ \/ _ \/ ___/ __ `/ __/ _ \/ __  /
//  _/ // / / / /_/  __/ / / / /_   / /_/ /  __/ / / /  __/ /  / /_/ / /_/  __/ /_/ /
// /___/_/ /_/\__/\___/_/ /_/\__/   \____/\___/_/ /_/\___/_/   \__,_/\__/\___/\__,_/

export interface CreateForceUser {
    name: string;
    side: Side;
    speciality: Speciality;
    lightSaberColor: LightSaberColor;
}
